tests/unit/test_runner.py

```diff
@@ -0,0 +1,140 @@
+import io
+import logging
+
+import pytest
+from west import runner
+from west import settings
+
+
+def test_clean_sql_code():
+    data = """
+        SELECT author_data.*
+        -- a comment
+        \\timing
+        FROM author_data;
+    """
+    expected = "\nSELECT author_data.*\nFROM author_data;\n\n"
+
+    assert runner.clean_sql_code(data) == expected
+
+
+def test_block_append_line():
+    block = runner.Block()
+    block.append_line("select true;")
+
+    assert block.content == "select true;"
+
+    block.append_line("commit;")
+
+    assert block.content == "select true;commit;"
+
+
+def test_block_append_line_closed():
+    block = runner.Block()
+    block.close()
+
+    with pytest.raises(runner.SQLRunnerException):
+        block.append_line("select true")
+
+
+def test_block_close():
+    block = runner.Block()
+    block.close()
+
+    assert block.closed is True
+
+
+def test_block_close_closed():
+    block = runner.Block()
+    block.close()
+
+    with pytest.raises(runner.SQLRunnerException):
+        block.close()
+
+
+def test_block_run(mocker, caplog):
+    caplog.set_level(logging.DEBUG)
+
+    # mock the db cursor
+    connection = mocker.MagicMock()
+    connection.cursor().__enter__().rowcount = 1
+
+    block = runner.Block()
+    block.append_line("select true;")
+    block.append_line("select false;")
+
+    count = block.run(connection)
+
+    assert count == 2
+
+    assert caplog.record_tuples == [
+        ('west.runner', 10, 'Running one statement... <<select true;>>'),
+        ('west.runner', 10, f"Affected 1 rows"),
+        ('west.runner', 10, 'Running one statement... <<select false;>>'),
+        ('west.runner', 10, f"Affected 1 rows"),
+    ]
+
+
+def test_simple_block_run(mocker):
+    connection = mocker.MagicMock()
+    execute = connection.cursor.return_value.__enter__.return_value.execute
+
+    block = runner.SimpleBlock()
+    block.append_line("select true")
+    block.run(connection)
+
+    execute.assert_called_with("select true\n")
+
+
+def test_meta_block_init():
+    with pytest.raises(runner.SQLRunnerException):
+        runner.MetaBlock("invalid command")
+
+
+def test_meta_block_run(mocker, caplog):
+    caplog.set_level(logging.DEBUG)
+
+    # mock the db cursor
+    connection = mocker.MagicMock()
+    connection.cursor().__enter__().rowcount = 0
```

Comment:
is really looks like you could use a fixture for this file, that would give you a connection, with a cursor attribute.

OR, maybe all those objects should accept a cursor parameter rather than a connection ? (not sure but maybe worth thinking about !

Reply:
I see you wen't along with the second solutions :) Is it changing something in the scope of transaction (does this make smaller or bigger transactions, or are transactions exactly as big as they used to be ?)

Reply:
Hm, reading the code and what the context manager does, you may want to add something in the for loop : 
http://initd.org/psycopg/docs/usage.html#with-statement

Transactions used to be closed (committed or rolled back) for every block, and now they are not, if I read this correctly. I'd say you want to keep the behaviour (but doing in with a single cursor is perfectly fine, and I think you'll agree it makes the code and tests much more readable)

Reply:
Looking at the code it seems that we're using psycopg in autocommit mode :
https://github.com/peopledoc/west/blob/37dce67b15647d99220975bb3c0c086491012a24/west/db.py#L43

So if I understand well no transaction are created and each `execute` statement is immediately committed.

Is it really what we want ?